File: 2503-longest-subarray-with-maximum-bitwise-and/longest-subarray-with-maximum-bitwise-and.py
class Solution:
    def longestSubarray(self, nums: list[int]) -> int:
        # Step 1: Find the maximum value
        max_val = max(nums)

        max_len = 1
        count = 0
        i = 0

        # Step 2: Iterate to find longest streak of max_val
        while i < len(nums):
            if nums[i] == max_val:
                while i < len(nums) and nums[i] == max_val:
                    count += 1
                    i += 1
                max_len = max(max_len, count)
                count = 0
            else:
                i += 1

        return max_len

File: 2503-longest-subarray-with-maximum-bitwise-and/longest-subarray-with-maximum-bitwise-and.go
func longestSubarray(nums []int) int {
	// Step 1: Find the maximum value
	maxVal := nums[0]
	for _, num := range nums {
		if num > maxVal {
			maxVal = num
		}
	}

	maxLen := 1
	count := 0
	i := 0

	// Step 2: Iterate to find longest streak of maxVal
	for i < len(nums) {
		if nums[i] == maxVal {
			for i < len(nums) && nums[i] == maxVal {
				count++
				i++
			}
			if count > maxLen {
				maxLen = count
			}
			count = 0
		} else {
			i++
		}
	}

	return maxLen
}
